framework/src/actions/CheckOutputAction.C

```diff
@@ -50,10 +50,10 @@ CheckOutputAction::checkVariableOutput(const std::string & task)
   {
     // Loop through the actions for the given task
     const std::vector<Action *> & actions = _awh.getActionsByName(task);
-    for (std::vector<Action *>::const_iterator it = actions.begin(); it != actions.end(); ++it)
+    for (auto act : actions)
```

Comment:
Funny you should bring that up since I'm currently going through the grain tracker trying various C++11 things out. I have a lot of loops that have `unsigned int`s and `dof_id_type`s in them and I've just been declaring the iterators as `auto foo` instead of `const auto foo &`. 

I don't know what I think. Now that I'm using auto I'm trying to be descriptive with the variable names so that I know the underlying type. I look at the loop and think, that's just POD or a raw pointer, so I can just copy it. In other cases, I'm declaring it as a reference. I'm leaning towards use your best judgement rather than laying down a hard rule. I bet there is no difference in timing, like virtually none so I can be convinced to go either way.